### PR TITLE
[main] Update vim to 8.2.3582 and atftp to 0.7.5

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Add vim updates up through 8.2.3582 and atftp to 0.7.5

Cherry-pick of 30e720dc353cc47da30ee7fe508e5402f14b10c9 8931d0b9b79fe92dc28c22494e7b21c9f3dea945 8abde10b54e527b1cbb1ab8157f54a6e79a5763d 8a11f88edb7720c794322f02f3f6191d5eef142c 0ba671d72dbc533a760b9bc9a9f900af45cbb37c d7a066de8cb5aab97296423175e69398a5ac421d

Signed-off-by: Chris Co <chrco@microsoft.com>

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-3778
- https://nvd.nist.gov/vuln/detail/CVE-2021-3796
- https://nvd.nist.gov/vuln/detail/CVE-2021-41054
- https://nvd.nist.gov/vuln/detail/CVE-2021-3875
- https://nvd.nist.gov/vuln/detail/CVE-2021-3872
- https://nvd.nist.gov/vuln/detail/CVE-2021-3903
- https://nvd.nist.gov/vuln/detail/CVE-2021-3927
- https://nvd.nist.gov/vuln/detail/CVE-2021-3928

### CODE REVIEW

File: SPECS/vim/vim.spec [hunk 1]
```diff
@@ -1,25 +1,25 @@
 %define debug_package %{nil}
 Summary:        Text editor
 Name:           vim
-Version:        8.1.1667
+Version:        8.2.3564
```
Comment (mfrw) [inline SPECS/vim/vim.spec:4]: Should we upgrade to `8.2.3582` instead; per #1635

**Review by mfrw — APPROVED** [review]
LGTM :)

File: SPECS/vim/vim.spec [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
+Version:        8.2.3582
+Release:        1%{?dist}
 License:        Vim
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Applications/Editors
 URL:            https://www.vim.org
-#Source0:       https://github.com/%{name}/%{name}/archive/v%{version}.tar.gz
-Source0:        %{name}-%{version}.tar.gz
+Source0:       https://github.com/%{name}/%{name}/archive/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/vim/vim.spec:11]: Nit-pick: could you aling the spaces with the other tags? Running a linter should fix all such issues.

File: SPECS/vim/vim.spec [hunk 3]
```diff
@@ -66,117 +63,126 @@ au BufEnter,BufNew *.py set tabstop=4 shiftwidth=4 expandtab
...
         call system('install -d -m 700 ~/.vim/swap')
 endif
-set directory=~/.vim/swap//
+set directory=~/.vim/swap/
 " End %{_sysconfdir}/vimrc
 EOF
 
 %check
 sed -i '/source test_recover.vim/d' src/testdir/test_alot.vim
-make test
```
Comment (PawelWMS) [inline SPECS/vim/vim.spec:75]: Could you add a short comment explaining what we're doing with this changes? I'm guessing we're disabling some invalid/irrelevant tests.

Comment (christopherco) [reply SPECS/vim/vim.spec:75]: Changes were a cherry-pick from 1.0-dev's vim. @chalamalasetty what was the reasoning behind this change?

**Pull request merged**